src/QirRuntime/lib/QIR/arrays.cpp

```diff
@@ -9,13 +9,29 @@
 #include <string.h> // for memcpy
 #include <vector>
 
+#include "allocationsTracker.hpp"
+#include "CoreTypes.hpp"
+#include "context.hpp"
 #include "qirTypes.hpp"
 #include "quantum__rt.hpp"
 
-#include "CoreTypes.hpp"
+namespace Microsoft
+{
+namespace Quantum
+{
+    extern thread_local std::unique_ptr<QirExecutionContext> g_context;
+}
+} // namespace Microsoft

```

Comment:
Minor:
I see this in 4 places. You may want to consider placing this in a header (`context.hpp`?).